### DISCUSSION TITLE
Create new command_history log for each session

### DISCUSSION THREAD

**Discussion by spauka** [microsoft/Qcodes#4798] [Ideas]
Currently, Qcodes saves the command history for each jupyter session running in a single file - usually stored at ~/.qcodes/command_history.log. When a session is restarted, the command history continues appended to the same file.

It might be preferrable to have a separate command_history log file created for each new jupyter session. In particular when running multiple consoles (e.g. if running analysis in one and measurement in the other), to have a separate history for each one rather than having them interleaved, making it difficult to reconstruct a session.

**Comment by jenshnielsen** [comment]
That sounds reasonable. The logic is here https://github.com/QCoDeS/Qcodes/blob/master/qcodes/logger/logger.py#L312 it should be possible to extend the filename with either a counter, thread/processid or something similar to that
